alibi_detect/saving/tensorflow/_loading.py

```diff
@@ -219,7 +181,7 @@ def load_embedding(src: str, embedding_type, layers) -> TransformerEmbedding:
 #######################################################################################################
 # TODO: Everything below here is legacy loading code, and will be removed in the future
 #######################################################################################################
-def load_detector_legacy(filepath: Union[str, os.PathLike], suffix: str, **kwargs) -> Detector:
+def load_detector_legacy(filepath: Union[str, os.PathLike], suffix: str, **kwargs) -> BaseDetector:
```

Comment:
Are we distinguishing between legacy detector formats and config detector formats via `BaseDetector` and `ConfigDector`? If so, isn't `BaseDetector` a bit misleading (as it's a base class; further, don't config-based detectors inherit from `BaseDetector` anyway?)?

Reply:
So the `ConfigDetector` class (Now the `ConfigurableDetector` protocol) distinguishes which detectors can be saved using the config approach. But basically, I'm just using it to check the detectors have the correct methods required for config-driven saving. In the case of legacy saving, it's not a question of what methods the detector has it's whether or not the function to save the detector exists. I can add a protocol for those detectors if we think that's a good idea?

Reply:
I'm not sure we need it unless there is a clear protocol definition of what constitutes a legacy detector? I'm just confused why the return type is `BaseDetector`, espcially because e.g. `MMDDrift` doesn't inherit from `BaseDetector`.

Reply:
Ah good point, does it make sense to define two protocols in that case? So:

```py
@runtime_checkable
class Detector(Protocol):
    meta: Dict

    def predict(self): ...


@runtime_checkable
class ConfigurableDetector(Detector):
    def get_config(self): ...

    def from_config(self): ...

    def _set_config(self): ...
```

Otherwise, we'll have to use a Union of all of them as there's no common base class. I think this will reintroduce the tensorflow import issues that we're trying to circumvent.

Reply:
Sounds reasonable to me if @ascillitoe agrees